scio-core/src/main/scala/com/spotify/scio/coders/KryoAtomicCoder.scala

```diff
@@ -23,15 +23,54 @@ import com.google.cloud.dataflow.sdk.coders.Coder.Context
 import com.google.cloud.dataflow.sdk.coders._
 import com.google.cloud.dataflow.sdk.util.VarInt
 import com.google.common.io.ByteStreams
+import com.google.common.reflect.ClassPath
 import com.google.protobuf.Message
 import com.twitter.chill._
 import com.twitter.chill.algebird.AlgebirdRegistrar
 import com.twitter.chill.protobuf.ProtobufSerializer
 import org.apache.avro.generic.GenericRecord
 import org.apache.avro.specific.SpecificRecordBase
+import org.slf4j.LoggerFactory
 
+import scala.collection.JavaConverters._
 import scala.collection.convert.Wrappers.JIterableWrapper
 
+@KryoRegistrar
+private class AlgebirdKryoRegistrar extends IKryoRegistrar {
+  override def apply(k: Kryo): Unit = new AlgebirdRegistrar()(k)
+}
+
+private object KryoRegistrarLoader {
+
+  private val logger = LoggerFactory.getLogger("KryoRegistrarLoader")
+
+  def load(k: Kryo): Unit = {
+    logger.info("Loading KryoRegistrars: " + registrars.mkString(", "))
+    registrars.foreach(_(k))
+  }
+
+  private val registrars: Seq[IKryoRegistrar] = {
+    logger.info("Initializing KryoRegistrars")
+    val classLoader = Thread.currentThread().getContextClassLoader
+    ClassPath.from(classLoader).getAllClasses.asScala.toSeq
+      .filter(_.getName.endsWith("KryoRegistrar"))
```

Comment:
How slow is it? What do you think of having all Kryo registrars declared in a package like `com.spotify.scio.kryo` and do `ClassPath.from(classLoader).getTopLevelClasses("com.spotify.scio.kryo")`? Either approach is a bit of a gotcha for users (even if documented), but with the package thing we wouldn't need the filter and it might be slightly less awkward (but maybe also more annoying?). And in either case maybe we should use an annotation processor to fail compilation (i.e. when class name does not end with KryoRegistrar) to eliminate the gotcha. Thoughts? @ravwojdyla